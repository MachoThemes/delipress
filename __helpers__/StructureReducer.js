export function getStructureState() { 
    return  {
        items: [
            {
                "rows" : [
                    {
                        "columns" : [
                            {
                                items: [
                                    {
                                        type: 1
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "columns" : [
                            {
                                items: [
                                    {
                                        type: 1
                                    }
                                ]
                            },
                            {
                                items: [
                                    {
                                        type :2
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "columns" : [
                            {
                                items: [
                                    {
                                        type: 4
                                    },
                                    {
                                        type: 3
                                    }
                                ]
                            }
                        ]
                    }
                ]
            },
            {
                "rows" : [
                    {
                        "columns" : [
                            {
                                type: 1
                            }
                        ]
                    },
                    {
                        "columns" : [
                            {
                                type: 2
                            }
                        ]
                    },
                    {
                        "columns" : [
                            {
                                type: 1
                            },
                            {
                                type: 1
                            }
                        ]
                    }
                ]
            }
        ]
    }
}