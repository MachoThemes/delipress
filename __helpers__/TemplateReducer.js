export function getTemplateState(){
    return {
        isLoading: false,
        template: null,
        items: null
    }
}