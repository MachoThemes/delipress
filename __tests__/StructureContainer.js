import React from 'react';
import ReactDOM from 'react-dom'
import { getStructureState } from '../__helpers__/StructureReducer'
import { getTemplateState } from '../__helpers__/TemplateReducer'

import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import StructureContainer from 'javascripts/react/modules/template/containers/StructureContainer'

const middlewares = [];
const mockStore   = configureStore(middlewares);

const getState    = {
    StructureReducer : getStructureState(),
    TemplateReducer : getTemplateState()
}

const store        = mockStore(getState)


describe('Structure container', () => {
    
    it('render structure container', () => {
        const wrapper = renderer.create(
            <Provider store={store}>
                <StructureContainer />
            </Provider>
        ).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

});
