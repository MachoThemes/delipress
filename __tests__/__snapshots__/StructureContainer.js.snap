exports[`Structure container render structure container 1`] = `
<div>
  <h1>
    Template your newsletter
  </h1>
  <div>
    <div
      style={
        Object {
          "margin": "5px",
          "width": "150px",
        }
      }>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "100%",
            }
          } />
      </div>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "50%",
            }
          } />
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "50%",
            }
          } />
      </div>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <button
      onClick={[Function]}>
      Choose this template
    </button>
  </div>
  <div>
    <div
      style={
        Object {
          "margin": "5px",
          "width": "150px",
        }
      }>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "100%",
            }
          } />
      </div>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "100%",
            }
          } />
      </div>
      <div
        className="row__preview"
        style={
          Object {
            "backgroundColor": "#ccc",
            "boxSizing": "border-box",
            "height": "40px",
          }
        }>
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "50%",
            }
          } />
        <div
          className="column__preview"
          style={
            Object {
              "backgroundColor": "#aaa",
              "border": "4px solid #ccc",
              "boxSizing": "border-box",
              "float": "left",
              "height": "38px",
              "width": "50%",
            }
          } />
      </div>
    </div>
    <button
      onClick={[Function]}>
      Choose this template
    </button>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    Go Customize newsletter
  </a>
</div>
`;
